core/vm/interpreter.go

```diff
@@ -215,10 +227,20 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool, i
 			// case of interrupting by timeout
 			select {
 			case <-interruptCtx.Done():
-				opcodeCommitInterruptCounter.Inc(1)
-				log.Warn("OPCODE Level interrupt")
+				txHash := interruptCtx.Value(InterruptedTxContext_currenttxKey).(common.Hash)
+				interruptedTxCache := interruptCtx.Value(InterruptedTxCacheKey).(*lru.Cache)
+
+				// if the tx is already in the cache, it means that it has been interrupted before and we will not interrupt it again
+				found, _ := interruptedTxCache.ContainsOrAdd(txHash, true)
+				if found {
+					interruptedTxCache.Remove(txHash)
```

Comment:
I might be wrong, but if we allow a previously interrupted transaction to continue its execution when the second time we see it, the node might end up with producing empty block again, and the transaction will still not be included in the block. Wouldn't this defeat the original purpose of interrupt commit?

Reply:
Moving this conversation to internal slack chat.